app/ui/src/app/integration/edit-page/step-configure/step-configure.component.ts

```diff
@@ -179,11 +138,34 @@ export class IntegrationStepConfigureComponent extends FlowPage implements OnIni
     }
     const prevStep = this.currentFlow.getPreviousStepWithDataShape(this.position);
     const nextStep = this.currentFlow.getSubsequentStepWithDataShape(this.position);
-    this.fetchDataShapesFor(prevStep, true)
-      .then(() => this.fetchDataShapesFor(nextStep, false))
+    this.currentFlow.fetchDataShapeFor(prevStep, true)
+      .then((inDataShape) => {
+        this.inputDataShape = inDataShape;
+        this.currentFlow.fetchDataShapeFor(nextStep, false)
+          .then((outDataShape) => {
+            this.outputDataShape = outDataShape;
+          })
+          .catch(response => this.handleDataShapeError(nextStep, response))
+      })
+      .catch(response => this.handleDataShapeError(prevStep, response))
       .then(() => this.loadFormSetup(step));
   }
 
+  private handleDataShapeError(step: Step, response: any) {
+    this.loading = false;
+    const error = JSON.parse(response['_body']);
```

Comment:
we'll need to double-check what excatly comes back in this error case, I'm not sure that the returned object is a `Response`, so `error` could wind up being undefined.

Reply:
Added a safeguard against NPE for now, let's see when test the error path.